public/editor/scripts/project-list.js

```diff
@@ -81,44 +81,88 @@ require(["jquery", "constants", "analytics", "moment"], function($, Constants, a
 
   $("#project-list").prepend(favoriteProjectsElementList);
 
+  var projectsToDelete = [];
+
   $(".project-delete").click(function() {
-    // TODO: we can do better than this, but let's at least make it harder to lose data.
-    if(!window.confirm("{{ deleteProjectConfirmationText }}")) {
-      return false;
+    if($(".delete-button").length === 0){
+      var deleteBtn = document.createElement("div");
+      deleteBtn.appendChild(document.createTextNode("Delete Projects"));
+      deleteBtn.className = "delete-button";
+      deleteBtn.onclick = function() {
+        // TODO: we can do better than this, but let's at least make it harder to lose data.
+        if(!window.confirm("{{ deleteProjectConfirmationText }}")) {
+          return false;
+        }
+        
+        Array.prototype.forEach.call(projectsToDelete, function(project){
```

Comment:
Based on 84 this is already an array, so we don't need to so prototype indirection. Also, if you're going to delete things in an array it is always safer to do so using reverse iteration, so taking those things together, and the knowledge that we want to end up with all the projects-to-delete having been deleted, let's rewrite this to something more concise:
```
while (projectsToDelete.length > 0) {
  deleteProject($(projectsToDelete.shift()), projectsToDelete.length === 0);
}
```
That's super short, and lets us define the "what should happen" code as its own function, which is better than defining it on the fly only for the duration of the iteration:
```
function deleteProject(project, lastProjectToDelete) {
  var projectId = project.data("id");
  var projectElementId = project.attr("id");
  // we're going to delete this element so event handlers should be garbage collected automatically
 
  analytics.event("DeleteProject");

  deleteProjectFromDB(projectId, lastProjectToDelete, onDeleteSuccess, onDeleteError);
}
```
This relies on three functions, `deleteProjectFromDB` which does the request part, handing off success to `onDeleteSuccess` and handing off errors to `onDeleteError`, making sure to pass `lastProjectToDelete` along as well, as it will let the code know whether it just deleted the last-project-in-the-list.